packages/shared/lib/core/profile/actions/active-profile/checkAndMigrateProfiles.ts

```diff
@@ -0,0 +1,69 @@
+import { get } from 'svelte/store'
+import { PROFILE_VERSION } from '../../constants'
+import { IPersistedProfile } from '../../interfaces'
+import { currentProfileVersion, profiles, saveProfile } from '../../stores'
+
+/**
+ * Migrates profile data in need of being modified to accommodate changes
+ * in a newer Firefly version.
+ */
+
+export function checkAndMigrateProfiles(): void {
+    const shouldMigratePersistedProfiles = (get(currentProfileVersion) ?? -1) < PROFILE_VERSION
+    if (shouldMigratePersistedProfiles) {
+        migratePersistedProfiles()
+        currentProfileVersion.set(PROFILE_VERSION)
+    }
+}
+
+function migratePersistedProfiles(): void {
+    const _profiles = get(profiles)
+    for (const profile of _profiles) {
+        migratePersistedProfile(profile)
+    }
+}
+
+function migratePersistedProfile(persistedProfile: IPersistedProfile): void {
+    let migratedPersistedProfile = persistedProfile
+    let migrationVersion = get(currentProfileVersion)
+    for (migrationVersion; migrationVersion < PROFILE_VERSION; migrationVersion++) {
+        migratedPersistedProfile = persistedProfileMigrationsMap?.[migrationVersion]?.(migratedPersistedProfile)
```

Comment:
we can remove the first `?` because it is always defined

Reply:
Which question mark? If someone it on a version or we skip a migration for a version then it will break truing to call this function . I think actually we should replace it with
```suggestion
        migratedPersistedProfile = persistedProfileMigrationsMap?.[migrationVersion]?.(migratedPersistedProfile) ?? migratedPersistedProfile 
```

Reply:
Updated